uniffi_bindgen/src/bindings/kotlin/gen_kotlin/names.rs

```diff
@@ -0,0 +1,92 @@
+/* This Source Code Form is subject to the terms of the Mozilla Publie
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
```

Comment:
I have a vague hand-wavy concern about factoring this away from the `CodeOracle` and into static helper functions - we lose the ability to do any config-based adjustments to the generated names. I know we currently do not support config-based adjustments to names, but it's been in the back of my mind as something that will come up at some point (see e.g. some of the proposals in https://github.com/mozilla/uniffi-rs/issues/400).

On the other hand, it does simplify things *a lot*, so I'm open to calling YAGNI on any clever name-generating shenanigans until it does actually come up in a concrete use-case.

@jhugman what do you think?